pytd/writer.py

```diff
@@ -49,7 +49,7 @@ def __init__(self, apikey, endpoint, td_spark_path=None, download_if_missing=Tru
         if 'eu01' in endpoint:
             site = 'eu01'
 
-        self.td_spark = self._fetch_td_spark(apikey, site, td_spark_path, download_if_missing)
+        self.td_spark = self._fetch_td_spark(apikey, site, td_spark_path, download_if_missing, endpoint)
```

Comment:
If we pass entire `endpoint` string to `_fetch_td_spark`, you can simply replace `site` arg with `endpoint` and set `site` value inside of the fetch method:

```py
def _fetch_td_spark(self, apikey, endpoint, td_spark_path, download_if_missing):
    site = # ...
```

Reply:
definitely. fixed